developing-key4hep-software/WritingAlgorithms.md

```diff
@@ -75,3 +297,14 @@ GDB console. To interrupt running of the Gaudi steering use `CTRL+C`.
 
 More details how to run GDB with Gaudi can be found in
 [LHCb Code Analysis Tools](https://twiki.cern.ch/twiki/bin/view/LHCb/CodeAnalysisTools#Debugging_gaudirun_py_on_Linux_w).
+
+## Avoiding const in `operator()`
+There is a way of working around `operator()` being const and that is by adding
+the keyword `mutable` to our data member. This will allow us to change our data
+member inside `operator()` and will cause code that wasn't compiling because of
+this to compile. Of course, this is not a good idea because unless the member of
+our class is thread-safe, that means that our algorithm is no longer thread-safe
+and running with multiple threads can cause different results. Even worse than
+that, it's very possible that there are not any errors or crashes but the
+results are simply wrong from having several threads changing a member at the
+same time, for example.
```

Comment:
I am not sure whether we should actually mention this. People who need to modify internal state will (most likely) know this, and people who don't know this probably shouldn't be using `mutable` ;) Maybe, there should rather be a paragraph explaining where setup work can be done (does Functional still have an `initialize`?)

Reply:
But hiding it won't make people not use mutable so I'd rather keep it so that if someone reads that at least they know what are the consequences, because people adding `mutable` so that the code compiles has already happened in experiments that use Gaudi. On top of that there will be places where it will be necessary to use `mutable`, like histogramming

Reply:
Fair enough.